### PR TITLE
Stats: Update Overview Components

### PR DESCRIPTION
**Pull request by timmyc**
This branch is part of the stats maintenance project and seeks to re-organize and cleanup the components that comprise the stats overview page.

__Files Moved__
- `stats-overview.jsx` -> `overview.jsx` - this is the main component for the overview page so leaving it at the root level in the stats directory
- `overview/` -> `stats-site-overview/` - this was always confusing name-wise.  This component builds out a single "card" of a site overview
- `module-site-overview-placeholder.jsx` -> `stats-overview-placeholder/index.jsx`

All files have been updated to ES6 syntax if applicable.

__To Test__
- Open up the [stats overview page](http://calypso.localhost:3000/stats) and verify everything looks the same as production
- Open up a stats insights page for a site, verify the "Today's Stats" section looks the same.

### CODE REVIEW

File: client/my-sites/stats/stats-site-overview/index.jsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+					<SiteIcon site={ site } size={ 24 } />
+				</div>
+			);
+
+			headerPath = route.getStatsDefaultSitePage( site.slug );
+		}
+
+		return (
+			<Card key={ site.ID } className="stats__overview stats-module is-site-overview">
+				<div className="module-header">
```
Comment (mtias) [inline client/my-sites/stats/stats-site-overview/index.jsx]: We should clean up all these classes.

Comment (mtias) [inline client/my-sites/stats/stats-site-overview/index.jsx]: Meaning the Card should have `.stats-site-overview` and the modules should be `.stats-site-overview__module-header` or create a `stats-module` component.

**Comment by timmyc** [comment]
@mtias I totally agree with your comments above, but right now we are doing the first step of moving these components into new directories and then are going to have follow-up PR's that break up the monolith `_stats.scss` and put styles into these new component directories.

The goal of doing it in this fashion is to keep the PR sizes smaller and more focused.  As-is this PR kind of got a bit larger than I was hoping.

**Comment by alternatekev** [comment]
LGTM :+1:

**Pull request merged**